Sming/Components/ssl/include/Network/Ssl/Session.h

```diff
@@ -0,0 +1,178 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * Session.h
+ *
+ ****/
+
+#pragma once
+
+#include "Context.h"
+#include "KeyCertPair.h"
+#include "Validator.h"
+#include <Platform/System.h>
+
+class TcpConnection;
+
+namespace Ssl
+{
+/**
+ * @brief Maximum Fragment Length Negotiation https://tools.ietf.org/html/rfc6066
+ *
+ * 0,1,2,3..6 corresponding to off,512,1024,2048..16384 bytes
+ *
+ * The allowed values for this field are: 2^9, 2^10, 2^11, and 2^12
+ *
+ */
+enum FragmentSize {
+	eSEFS_Off, ///< Let SSL implementation decide
+	eSEFS_512, ///< 512 bytes
+	eSEFS_1K,  ///< 1024 bytes
+	eSEFS_2K,
+	eSEFS_4K,
+	eSEFS_8K,
+	eSEFS_16K,
+};
```

Comment:
Still niggles. The best I can think of is this:

```
enum class FragmentSize {
	Default, ///< Let SSL implementation decide
	B512, ///< 512 bytes
	K1,  ///< 1024 bytes
	K2,
	K4,
	K8,
	K16,
};
```

Reply:
When I see this, I'm imagining that FramentSize today refers to the max_fragment_length of today (https://tools.ietf.org/html/rfc6066) and in the future, being library dependent as they update, will refer to record_size_limit of tomorrow (https://tools.ietf.org/html/rfc8449).  Record size limit can be any value where fragment size is only an enumeration of valid sizes.  The enumeration of sizes looks granular enough to be to be usable and would translate just fine between the transition.  If I were to split hairs, the enumeration name isn't generic between the two standards -- your call.

Reply:
At the application level, what information should this convey? If the library handles the detail, my thought is that we just want to limit the buffer size  so we have some control over RAM usage. That being the case, we'd call this `maxBufferSize` and it would just be a hint. Does that cover it, or am I missing something?

Reply:
At the application level, what information should this convey? If the library handles the detail, my thought is that we just want to limit the buffer size  so we have some control over RAM usage. That being the case, we'd call this `maxBufferSize` and it would just be a hint. Does that cover it, or am I missing something?

Reply:
> maxBufferSize ... K1,  ///< 1024 bytes

Agreed, make a PR and I will merge it.